dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ResourceLinkResource.java

```diff
@@ -120,28 +134,38 @@ private Contentlet getContentlet(final String inode,
                                      final PageMode pageMode) throws DotDataException, DotSecurityException {
 
         Contentlet contentlet = null;
-        PageMode mode         = pageMode;
+        try {
+            PageMode mode = pageMode;
 
-        if(UtilMethods.isSet(inode)) {
+            if (UtilMethods.isSet(inode)) {
 
-            Logger.debug(this, ()-> "Finding the contentlet by inode: " + inode);
-            contentlet = this.contentletAPI.find
-                    (inode, initDataObject.getUser(), mode.respectAnonPerms);
+                Logger.debug(this, () -> "Finding the contentlet by inode: " + inode);
+                contentlet = this.contentletAPI.find
+                        (inode, initDataObject.getUser(), mode.respectAnonPerms);
 
-            DotPreconditions.notNull(contentlet, ()-> "contentlet-was-not-found", DoesNotExistException.class);
-        } else if (UtilMethods.isSet(identifier)) {
+                DotPreconditions.notNull(contentlet, () -> "contentlet-was-not-found", DoesNotExistException.class);
+            } else if (UtilMethods.isSet(identifier)) {
 
-            Logger.debug(this, ()-> "Finding the contentlet by identifier: " + identifier);
-            mode = PageMode.EDIT_MODE; // when asking for identifier it is always edit
-            final Optional<Contentlet> currentContentlet =  language <= 0?
-                    Optional.ofNullable(this.contentletAPI.findContentletByIdentifier(identifier, mode.showLive,
-                            sessionLanguage.get(), initDataObject.getUser() , mode.respectAnonPerms)):
-                    this.contentletAPI.findContentletByIdentifierOrFallback
-                            (identifier, mode.showLive, language, initDataObject.getUser(), mode.respectAnonPerms);
+                Logger.debug(this, () -> "Finding the contentlet by identifier: " + identifier);
+                mode = PageMode.EDIT_MODE; // when asking for identifier it is always edit
+                final Optional<Contentlet> currentContentlet = language <= 0 ?
+                        Optional.ofNullable(this.contentletAPI.findContentletByIdentifier(identifier, mode.showLive,
+                                sessionLanguage.get(), initDataObject.getUser(), mode.respectAnonPerms)) :
+                        this.contentletAPI.findContentletByIdentifierOrFallback
+                                (identifier, mode.showLive, language, initDataObject.getUser(), mode.respectAnonPerms);
 
-            DotPreconditions.isTrue(currentContentlet.isPresent(), ()-> "contentlet-was-not-found", DoesNotExistException.class);
-            contentlet = currentContentlet.get();
+                DotPreconditions.isTrue(currentContentlet.isPresent(), () -> "contentlet-was-not-found", DoesNotExistException.class);
+                contentlet = currentContentlet.get();
 
+            }
+        } catch (DoesNotExistException e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A catch statement that catches an exception only to rethrow it should be avoided.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5644246)